version: '3.9'

services:
  bot_consumer:
    build:
      context: .
      dockerfile: Dockerfile_consumer_bot
    ports:
      - "1097:1097"
    restart: on-failure
    depends_on:
      - rabbit
    links:
      - rabbit

  rabbit:
    build:
      context: rabbit
      dockerfile: Dockerfile
    ports:
      - "5672:5672"

  db:
    image: postgres:13-alpine
    ports:
      - "5432:5432"
    environment:
      - POSTGRES_DB=config
      - POSTGRES_USER=postgresql
      - POSTGRES_PASSWORD=postgresql

  psychologists_apis:
    build:
      context: .
      dockerfile: Dockerfile_apis
    ports:
      - 8001:8001
    command: bash -c "uvicorn app.main:app --host 0.0.0.0 --port 8001"
    restart: on-failure
    depends_on:
      - rabbit
      - db
    links:
      - rabbit
      - db
    volumes:
      - media:/media

  psychologists_bot:
    build:
      context: .
      dockerfile: Dockerfile_psychologists_bot
    restart: on-failure
    depends_on:
      - db
    links:
      - db
    volumes:
      - media:/media


  admin_panel:
    build:
      context: .
      dockerfile: Dockerfile_admin_panel
    ports:
      - 8000:8000
    command: python3 manage.py runserver 0.0.0.0:8000
    restart: on-failure
    depends_on:
      - db
    links:
      - db
    volumes:
      - media:/media

volumes:
  media:
